File: ManagerData/DataModels/TaskHistory.cs
using ManagerData.Constants;

namespace ManagerData.DataModels;

public class TaskHistory : BaseDataModel
{
    public Guid TaskId { get; set; }
    public int? SourceStatusId { get; set; }
    public int? DestinationStatusId { get; set; }
    public Guid InitiatorId { get; set; }
    public Guid? TargetMemberId { get; set; }
    public TaskActionType ActionType { get; set; }
    
    public TaskDataModel Task { get; set; }
    public MemberDataModel Initiator { get; set; }
    public MemberDataModel TargetMember { get; set; }
}

File: ManagerView/src/app/components/models/task-history.ts
import { Member } from "./member";

export interface TaskHistory {    
    taskId : string;
    sourceStatusId : number;
    destinationStatusId : number;
    initiatorId : string;
    targetMemberId : string;
    actionType : TaskActionType;
    createdAt : Date;
    description : string;
    name : string;
    initiator? : Member;
    targetMember? : Member;
}

export enum TaskActionType {
    Created = 0,
    StatusChanged,
    Assigned,
    Reassigned,
    MemberAdded,
    Commented,
    Renamed,
    DescriptionChanged,
    DeadlineChanged,
    Deadline,
    FileAdded,
    FileRemoved
}

export enum TaskActionHeader {
    Created = "Создание задачи",
    StatusChanged = "Статус изменен",
    Assigned = "Задача прикреплена",
    Reassigned = "Задача откреплена",
    MemberAdded = "Исполнитель добавлен",
    Commented = "Комментарий добавлен",
    Renamed = "Переименование задачи",
    DescriptionChanged = "Обновление описания",
    DeadlineChanged = "Смена крайнего срока",
    Deadline = "Наступил крайний срок",
    FileAdded = "Добавлен файл",
    FileRemoved = "Удалён файл",
}